### PR TITLE
enable wshadow

### LINKED ISSUE #22714: Enable Wshadow flag
Need to set the flag and fix all the violations. Submodules should be done first

### PR DESCRIPTION
**Pull request by mayfieldiv**
Resolves #22714
Not sure how strict of naming conventions y'all want when there's a collision between a parameter and a class member (e.g. `ack_` vs `ack`, `init_rgb_type` vs `rgb_type`, `state` vs `s`)

Blocked by:
- [x] https://github.com/commaai/cereal/pull/214
- [x] https://github.com/commaai/rednose/pull/21

Related:
- [x] https://github.com/commaai/opendbc/pull/461

### CODE REVIEW

**Comment by pd0wm** [comment]
Can you add the flag to the cereal sconstruct so we catch any errors immediately? Same for rednose and opendbc.

File: selfdrive/ui/qt/maps/map.cc [hunk 1]
```diff
@@ -576,9 +576,9 @@ void MapInstructions::updateDistance(float d) {
   distance->setText(distance_str);
 }
 
-void MapInstructions::showError(QString error) {
+void MapInstructions::showError(QString error_text) {
```
Comment (pd0wm) [inline selfdrive/ui/qt/maps/map.cc:579]: This was an actual bug!

Comment (pd0wm) [reply selfdrive/ui/qt/maps/map.cc:579]: Ran into it in https://github.com/commaai/openpilot/pull/22767

Comment (mayfieldiv) [reply selfdrive/ui/qt/maps/map.cc:579]: haha good to hear!

**Pull request merged**